src/ingest-pipeline/misc/tools/survey.py

```diff
@@ -198,15 +211,46 @@ def __init__(self, prop_dct, entity_factory):
             self.data_types = f'{self.data_types}'
             self.notes.append('BAD TYPE NAME')
             self.is_derived = None
-        self.donor_uuid = prop_dct['donor']['uuid']
+        self._donor_uuid = None  # requires graph search, so be lazy
         self.group_name = prop_dct['group_name']
         c_h_g = prop_dct['contains_human_genetic_sequences']
         if isinstance(c_h_g, str):
             c_h_g = (c_h_g.lower() in ['yes', 'true'])
         self.contains_human_genetic_sequences = c_h_g
         self._kid_dct = None
         self._parent_dct = None
-    
+        self._organs = None
+
+    @property
+    def donor_uuid(self):
```

Comment:
I don't quite understand the difference in this if/else case (probably b/c I don't know how this data is structured). Is this meant to be able to handle the case where a parent entity is not marked as a Donor, but does have a donor_uuid associated with it?

Reply:
Ah.  That's awkwardly coded, but the point is,
* if the parent is a Donor, it's uuid is the thing we want.
* if not, the parent is a Dataset or a Sample, in which case evaluating parent_entity.donor_uuid (with no leading underscore) will recursively call its donor_uuid property, causing it to check *its* parent.  So it's recursion with a property call each time, until it reaches the Donor and the first clause executes.  But we fill in all the _donor_uuid properties on the way down, so they are available in the cache.